pkg/phase.go

```diff
@@ -47,7 +49,17 @@ func (p *Phase) Objects() map[string]crv1alpha1.ObjectReference {
 
 // Exec renders the argument templates in this Phase's Func and executes with
 // those arguments.
-func (p *Phase) Exec(ctx context.Context, bp crv1alpha1.Blueprint, action string, tp param.TemplateParams) (map[string]interface{}, error) {
+func (p *Phase) Exec(ctx context.Context, cli crclientv1alpha1.CrV1alpha1Interface, as *crv1alpha1.ActionSet, bp crv1alpha1.Blueprint, action string, tp param.TemplateParams) (map[string]interface{}, error) {
```

Comment:
Should these be part of Phase struct instead? Could be reusable later if needed for other functions as well.

Reply:
mmm, I don't think we have a lot of methods of phase that is going to need the actionset. Also, actionset is not a property of phase, that is the reason I decided not to include that in the Phase struct.
So, if we add this as field of Phase, everywhere we create phase, we will have to either somehow try to init actionset or init it as `nil`. And we wont have actionset everywhere we init phase.

Reply:
So it can be nil by default wherever we won't have actionset. We right now anyway have to pass `nil` while making `phase.Exec` call.
But also makes sense since ActionSet is not a property of field.
We should still move cli to phase struct right?

Reply:
Hi @ankitjain235,
Like we discussed offline I think it would be a good idea if we update the actionset right before calling `Exec` for phase.